xilem_core/src/element.rs

```diff
@@ -82,3 +82,15 @@ where
     /// Replace the inner value of this reference entirely
     fn replace_inner(this: Self::Mut<'_>, child: Child) -> Self::Mut<'_>;
 }
+
+/// Element type for views which don't impact the element tree.
+///
+/// Views with this element type can be included in any [`ViewSequence`](crate::ViewSequence) (with the
+/// correct `State` and `Action` types), as they do not need to actually add an element to the sequence.
+///
+/// These views can also as the `alongside_view` in [`fork`](crate::fork).
+pub struct NoElement;
```

Comment:
The reason that `NoElement` exists is so that it can be produced from nothing, and so can be inserted anywhere into the view tree.

I don't see how it can be meaningful for `NoElement` to contain data?

Reply:
I mean certainly not for `NoElement`, but a view could create e.g. an `AsyncElement` which may contain state relevant to that element, or something along those lines, which is managed by a `View` such as `fork`, i.e. having a custom view runner for different elements inside such a view. But right now, it probably doesn't make sense.